include/albatross/src/utils/block_utils.hpp

```diff
@@ -302,6 +312,51 @@ inline Eigen::Matrix<_Scalar, _Rows, _Cols> BlockDiagonalLDLT::sqrt_solve(
   return output;
 }
 
+inline std::map<size_t, Eigen::Index>
+BlockDiagonalLDLT::block_to_row_map() const {
+  Eigen::Index row = 0;
+  std::map<size_t, Eigen::Index> block_to_row;
+
+  for (size_t i = 0; i < blocks.size(); ++i) {
+    block_to_row[i] = row;
+    row += blocks[i].rows();
+  }
+  assert(row == cols());
+
+  return block_to_row;
+}
+
+template <class _Scalar, int _Rows, int _Cols>
+inline Eigen::Matrix<_Scalar, _Rows, _Cols> BlockDiagonalLDLT::async_solve(
+    const Eigen::Matrix<_Scalar, _Rows, _Cols> &rhs) const {
+  assert(cols() == rhs.rows());
+  Eigen::Matrix<_Scalar, _Rows, _Cols> output(rows(), rhs.cols());
+  auto solve_and_fill_one_block = [&](const size_t i, const Eigen::Index row) {
+    const auto rhs_chunk = rhs.block(row, 0, blocks[i].rows(), rhs.cols());
```

Comment:
nit: I'm curious what the difference between this and directly doing
```
output.block(row, 0, blocks[i].rows(), rhs.cols()) = blocks[i].solve(rhs.block(row, 0, blocks[i].rows(), rhs.cols()));
```
would be.  On one hand I think the type for `rhs_chunk` will be some sort of `Eigen` block view, so it might be pretty lightweight, but there *could* be an advantage to passing it directly into `solve` as an rvalue.  I really don't know if that'll even make a difference though, feel free to skip.